### PR TITLE
npcx: ps2: fix compiler error and Kconfig

### PR DESCRIPTION
**Pull request by ChiHuaL**
This PR:
1.  Fixes the compiler error by add the miss header file kernel.h for npcx PS/2 driver.
2.  Modify Kconfig `PS2_NPCX` has dependency on the PS2 device node's status for npcx PS2 Kconfig.
3.  Modify the PS2 sample code to use alias to select PS2 device instead of choosing a device of a specific chip's COMPAT.

### CODE REVIEW

File: samples/drivers/ps2/boards/mec1501modular_assy6885.overlay [hunk 1]
```diff
@@ -13,3 +13,9 @@
 	wakerx-gpios = <MCHP_GPIO_DECODE_155 GPIO_ACTIVE_HIGH>;
 	wakeup-source;
 };
+
+/ {
+	aliases {
+		ps2-port-0 = &ps2_0;
```
Comment (fabiobaltieri) [inline samples/drivers/ps2/boards/mec1501modular_assy6885.overlay:19]: hey, looks like most aliases have no dash before the number for some reasons, could you change this to `ps2-port0`?

**Review by fabiobaltieri — DISMISSED** [review]

File: samples/drivers/ps2/boards/mec1501modular_assy6885.overlay (hunk 1 above)
Comment (ChiHuaL) [reply samples/drivers/ps2/boards/mec1501modular_assy6885.overlay]: Done.

File: samples/drivers/ps2/boards/npcx7m6fb_evb.overlay [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+/ {
+	aliases {
+		ps2-port0 = &ps2_channel0;
+	};
+};
+
+&ps2_channel0 {
+       status = "okay";
```
Comment (keith-zephyr) [inline samples/drivers/ps2/boards/npcx7m6fb_evb.overlay:14]: This line is missing the hard tab.

File: samples/drivers/ps2/boards/npcx9m6f_evb.overlay [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+/ {
+	aliases {
+		ps2-port0 = &ps2_channel0;
+	};
+};
+
+&ps2_channel0 {
+       status = "okay";
```
Comment (keith-zephyr) [inline samples/drivers/ps2/boards/npcx9m6f_evb.overlay:14]: Needs a hard tab.

File: samples/drivers/ps2/boards/npcx7m6fb_evb.overlay (hunk 2 above)
Comment (ChiHuaL) [reply samples/drivers/ps2/boards/npcx7m6fb_evb.overlay]: Done.

File: samples/drivers/ps2/boards/npcx9m6f_evb.overlay (hunk 3 above)
Comment (ChiHuaL) [reply samples/drivers/ps2/boards/npcx9m6f_evb.overlay]: Done.

**Review by fabiobaltieri — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**